executor/grant.go

```diff
@@ -77,9 +77,20 @@ func (e *GrantExec) Next(ctx context.Context, req *chunk.Chunk) error {
 		dbNameStr := model.NewCIStr(dbName)
 		schema := e.ctx.GetInfoSchema().(infoschema.InfoSchema)
 		tbl, err := schema.TableByName(dbNameStr, model.NewCIStr(e.Level.TableName))
-		// Allow GRANT on non-existent table, see issue #28533
-		if err != nil && !terror.ErrorEqual(err, infoschema.ErrTableNotExists) {
-			return err
+		// Allow GRANT on non-existent table with at least create privilege, see issue #28533 #29268
+		if err != nil {
```

Comment:
Will this pr cause some compatibility problems?

Reply:
IMHO it won't as it is based on previous PR https://github.com/pingcap/tidb/pull/28882, and both of them have not delivered to release branch. Or we can wait on @morgo's opinion.

Reply:
I agree with you @unconsolable. It will only cause compatibility issues between from an unreleased version.